conftest.py

```diff
@@ -51,3 +55,47 @@ def pytest_itemcollected(item):
 
     if isinstance(item, pytest.DoctestItem):
         item.add_marker(doctest_marker)
+
+
+def pytest_addoption(parser: pytest.Parser) -> None:
+    """Hook to register custom CLI options and config values"""
+    parser.addoption(
+        "--strict-warnings",
+        action="store_true",
+        default=False,
+        help="Turn most warnings into errors",
+    )
+
+    parser.addini(
+        "filterwarnings_when_strict",
+        "Filters to apply after `-Werror` when --strict-warnings is active",
+        type="linelist",
+        default=[],
+    )
+
+
+def pytest_collection_modifyitems(
+    session: pytest.Session, config: pytest.Config, items: Iterable[pytest.Item]
+):
+    if not config.getoption("--strict-warnings"):
+        return
+
+    warning_filters = [
+        r"error",
```

Comment:
I don't think the `r` is necessary here

Reply:
True. I had it for consistency when things where hardcoded here.